### PR TITLE
HIVE-27472: Iceberg: Disallow creation of temporary tables

### PR DESCRIPTION
**Pull request by deniskuzZ**
### What changes were proposed in this pull request?

Disallow the creation of a Temporary Hive Iceberg tables

### Why are the changes needed?

Atlas by default skips ingesting temporary tables created in Hive. But when searched for the newly created temporary Hive tables, it displays them in the search results.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Unit test

### CODE REVIEW

**Review by simhadri-g — APPROVED** [review]
LGTM +1 (non-binding)
Please resolve merge confilts.

**Review by ayushtkn — APPROVED** [review]
LGTM

File: iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java [hunk 1]
```diff
@@ -165,6 +165,9 @@ public HiveIcebergMetaHook(Configuration conf) {
 
   @Override
   public void preCreateTable(org.apache.hadoop.hive.metastore.api.Table hmsTable) {
+    if (hmsTable.isTemporary()) {
+      throw new UnsupportedOperationException("Creation of iceberg temporary tables is not supported.");
```
Comment (aturoczy) [inline iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java:169]: I would change the order iceberg temporary table.  Also 
_"Creation of temporary iceberg tables is not supported"_

**Review by aturoczy — APPROVED** [review]

File: iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java (hunk 1 above)
Comment (aturoczy) [reply iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java:169]: Just a minor stuff.

Comment (deniskuzZ) [reply iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java]: changed

**Pull request merged**